CommandLine/emake/Main.cpp

```diff
@@ -34,7 +71,37 @@ int main(int argc, char* argv[])
     bool _run = options.GetOption("run").as<bool>();
     if (!_run) plugin.HandleGameLaunch();
 
-    return plugin.BuildGame(game.ConstructGame(), mode, options.GetOption("output").as<std::string>().c_str());
+    Game game;
+    std::string input_file = options.GetOption("input").as<std::string>();
+    if (input_file.size()) {
+      if (input_file.size() < 4 || input_file[input_file.length() - 4] != '.') {
+        std::cerr << "Error: Unknown filetype: can't read input file \""
+                  << input_file << "\"." << std::endl;
+        return 1;
+      }
+      std::string ext = tolower(input_file.substr(input_file.length() - 3, 3));
```

Comment:
not all files have 3 letters as extensions. use something like find_last_of(".") instead

Reply:
I was more concerned with someone adding something like 3fgz support later.